### ISSUE TITLE
SQLite blob literal syntax fails LT01

### ISSUE THREAD

**Issue by jasonaowen** [sqlfluff/sqlfluff#7300]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting a SQLite SQL file that uses [blob literal syntax](https://www.sqlite.org/lang_expr.html#literal_values_constants_) reports a `layout.spacing` violation, and fixing it results in SQLite not being able to parse the file.

> BLOB literals are string literals containing hexadecimal data and preceded by a single "x" or "X" character. Example: X'53514C697465'

### Expected Behaviour

SQLFluff should understand and allow SQLite blob literals.

### Observed Behaviour

```
$ sqlfluff parse --dialect sqlite blob-literal.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    expression:
[L:  1, P:  8]      |                        data_type:
[L:  1, P:  8]      |                            data_type_identifier:             'x'
[L:  1, P:  9]      |                        quoted_literal:                       "'53514C697465'"
[L:  1, P: 23]      |                [META] dedent:
[L:  1, P: 23]      |    statement_terminator:                                     ';'
[L:  1, P: 24]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:

$ sqlfluff fix --dialect sqlite blob-literal.sql
==== finding fixable violations ====
== [blob-literal.sql] FAIL                                                                                                    
L:   1 | P:   9 | LT01 | Expected single whitespace between data type identifier
                       | and quoted literal. [layout.spacing]
== [blob-literal.sql] FIXED
1 fixable linting violations found

$ sqlite3 < blob-literal.sql 
Parse error near line 1: no such column: x
  SELECT x '53514C697465';
         ^--- error here
```

### How to reproduce

`blob-literal.sql`:

```sql
SELECT x'53514C697465';
```

### Dialect

SQLite

### Version

sqlfluff, version 3.5.0

Python 3.12.3

### Configuration

No custom configuration; default rules only.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by franloza** [comment]
I opened a PR to fix this issue: https://github.com/sqlfluff/sqlfluff/pull/7332
